src/v/cloud_storage/cache_service.h

```diff
@@ -47,10 +51,30 @@ class cache {
     /// Remove element from cache by key
     ss::future<> invalidate(const std::filesystem::path& key);
 
+    // Total cleaned is exposed for better testability of eviction
+    uint64_t get_total_cleaned();
+
 private:
+    /// Triggers directory walker, creates a list of files to delete and deletes
+    /// them until cache size <= _cache_size_low_watermark * max_cache_size
+    ss::future<> clean_up_cache();
+
+    /// Triggers directory walker, creates a list of files to delete and deletes
+    /// only tmp files that are left from previous Red Panda run
+    ss::future<> clean_up_at_start();
+
+    bool is_tmp_file(const ss::sstring& filename);
+
     std::filesystem::path _cache_dir;
+    size_t _max_cache_size;
+    ss::lowres_clock::duration _check_period;
+
     ss::gate _gate;
     uint64_t _cnt;
+    static constexpr long double _cache_size_low_watermark{0.8L};
```

Comment:
why long double?

Reply:
Cause I multiply it by `uint64_t` and if I use `double`, it will loose precision.

Reply:
That shouldn't be a problem, the `uint64_t` value will hold a number of bytes on disk, let's say that the cache is 1PiB (which is very unrealistic), this value can be represented using regular `double` without precision loss (since it can represent 52-bit integers without loss). So I would say that `double` is more than good enough for the task.

Reply:
Yep, but clangd is complaining about `narrowing conversion from 'size_t' to double`. Will store it as `double` and convert it to `long double` before multiplying